### PR TITLE
feat(iterators): add `GuildIterator.with_counts`

### PR DESCRIPTION
**Pull request by spifory**
## Summary

Implements discord/discord-api-docs#5628

In case you're wondering, "why is the default for `with_counts` `False`?"
Well that is because the default on the documentation `false`, so just for the sake of consistency, that is why the default is `False`.

## This is a **Code Change**

- [ ] I have tested my changes.
- [x] I have updated the documentation to reflect the changes.
- [x] I have run `task pyright` and fixed the relevant issues.

### CODE REVIEW

**Comment by spifory** [comment]
I accidentally made this branch off of another branch, and not master, so that is the reason for the force-pushes

File: nextcord/iterators.py [hunk 1]
```diff
@@ -625,6 +625,11 @@ class GuildIterator(_AsyncIterator["Guild"]):
         The client to retrieve the guilds from.
     limit: :class:`int`
         Maximum number of guilds to retrieve.
+    with_counts: :class:`bool`
+        Whether to include approximate member and presence counts for the guilds.
+        Defaults to ``True``.
```
Comment (teaishealthy) [inline nextcord/iterators.py:630]: I don't think so
```suggestion
        Defaults to ``False``.
```

**Review by teaishealthy — CHANGES_REQUESTED** [review]

**Review by teaishealthy — APPROVED** [review]
is probably ok :rocket:

**Review by EmreTech — CHANGES_REQUESTED** [review]

File: nextcord/iterators.py [hunk 2]
```diff
@@ -706,20 +713,28 @@ async def _retrieve_guilds(self, retrieve: int) -> List[GuildPayload]:
         """Retrieve guilds and update next parameters."""
         raise NotImplementedError
 
-    async def _retrieve_guilds_before_strategy(self, retrieve: int) -> List[GuildPayload]:
+    async def _retrieve_guilds_before_strategy(
+        self, retrieve: int, with_counts: bool = False
```
[the remark this answers is not in the document]
Comment (spifory) [reply nextcord/iterators.py:717]: I wasn't thinking that it couldn't be used in both, I just figured I might have to add the parameter to that function as well.

**Review by EmreTech — CHANGES_REQUESTED** [review]
This PR fails testing. I got an [exception](https://workbin.dev/?id=1691938674796779) that yarl (which aiohttp uses for query parameters) doesn't support booleans.

You have to do something like `HTTPClient.get_guild`:
https://github.com/nextcord/nextcord/blob/a4bd5d2482614e8098c180c0e69100a0e5317b4d/nextcord/http.py#L1360

**Comment by spifory** [comment]
@EmreTech https://github.com/nextcord/nextcord/pull/1093/commits/244899b6ccd99e2556564a871648a95f53ab9014

**Review by EmreTech — APPROVED** [review]
Tested.

**Review by ooliver1 — APPROVED** [review]

**Pull request merged**